### DISCUSSION TITLE
spack-stack via singularity

### DISCUSSION THREAD

**Discussion by benjamin-cash** [JCSDA/spack-stack#1175] [Q&A]
Hello,

I have what I think is probably a basic containers question I was hoping someone could help me with. I am trying to adapt a set of instructions for using a container with **apptainer** on Frontera to **singularity** on my local cluster. In the Frontera script, there is the following command:
 
`source /root/.bashenv`
 
This command returns a permission error if run from the command line, but works just fine if run as part of the `apptainer exec` command. However, on my local cluster, trying to use this command in the script via `singularity exec` returns a permission error. Leaving it out results in a large number of errors related to not finding /opt and other paths related to modules.

I haven't been able to find any documentation on why this command works in one instance but not the other. Any suggestions would be greatly appreciated!

**Comment by climbfuji** [comment]
Ben, this is not a spack-stack question and I don't know the answer to your question, unfortunately. We build docker containers with spack-stack and convert them to singularity afterwards.

**Comment by benjamin-cash** [comment]
Hi @climbfuji  - Is there documentation on how to build the model using spack-stack via singularity? As a related question, would it be possible to get some of the other environments built as containers? For example, global-workflow requires gsi-addon for full functionality, but that doesn't seem to be one of the containers [here](https://hub.docker.com/u/noaaepic).

**Reply by climbfuji** [reply]
Yes to the latter - defer to EPIC/EMC for helping with that. One needs to create a different list of specs (currently there is only jedi-ci) and then use that with the container templates when running `spack stack create ctr`. All in readthedocs.

About building the model: again, defer to EPIC/EMC. Basically just need to source the entrypoint.sh script, maybe set one or two additional environment variables and run `build.sh`. Tagging @ulmononian and @AlexanderRichert-NOAA for awareness. I know that EPIC did experiment with the spack-stack container recipes, but I've never heard back/saw a spec yaml file being contributed back to the repo.

**Comment by benjamin-cash** [comment]
We got the issues ironed out so I will go ahead and close this.
